cpp/src/io/comp/debrotli.cu

```diff
@@ -1393,9 +1396,9 @@ static __device__ void HuffmanTreeGroupAlloc(debrotli_state_s *s, debrotli_huff_
     uint32_t ntrees         = group->num_htrees;
     uint32_t max_table_size = kMaxHuffmanTableSize[(alphabet_size + 31) >> 5];
     uint32_t code_size      = sizeof(uint16_t) * ntrees * max_table_size;
-    group->htrees[0]        = reinterpret_cast<uint16_t *>(local_alloc(s, code_size));
+    group->htrees[0]        = static_cast<uint16_t *>(local_alloc(s, code_size));
     if (!group->htrees[0]) {
-      if (s->fb_base) { group->htrees[0] = reinterpret_cast<uint16_t *>(s->fb_base + s->fb_size); }
+      if (s->fb_base) { group->htrees[0] = static_cast<uint16_t *>(s->fb_base) + s->fb_size / 2; }
```

Comment:
Where did the `/ 2` come from?

Reply:
`s->fb_base` used to be `uint8_t*`, but now it's `void*`. So now we have to cast before adding offset. We need a `uint16_t` pointer here, so I cast directly to `uint16_t*` here and halved the offset since sizeof of each element is doubled.